### PR TITLE
Fixes an edge case revelead on a test failure

### PR DESCRIPTION
**Pull request by cquiroz**

### CODE REVIEW

File: modules/core/shared/src/main/scala/lucuma/core/math/validation/MathValidators.scala [hunk 1]
```diff
@@ -53,7 +53,7 @@ trait MathValidators {
       a =>
         f"${truncateAngle(a).toSignedBigDecimalDegrees}%.2f"
           .replace("-0.00", "0.00")
-          .replace("180.00", "-180.00")
+          .replace("$180.00", "-180.00")
```
Comment (toddburnside) [inline modules/core/shared/src/main/scala/lucuma/core/math/validation/MathValidators.scala:56]: What is the `$` for? Is there actually a `$` in the original string?

Comment (toddburnside) [reply modules/core/shared/src/main/scala/lucuma/core/math/validation/MathValidators.scala:56]: Ah. I guess the old replace was changing "-180.00" to "--180.00"? But, the new replace doesn't seem to fix that, to me.

Comment (cquiroz) [reply modules/core/shared/src/main/scala/lucuma/core/math/validation/MathValidators.scala:56]: exactly the `--180` triggered a test error. `replace` takes a regex

Comment (toddburnside) [reply modules/core/shared/src/main/scala/lucuma/core/math/validation/MathValidators.scala:56]: Ah. OK. I didn't know `replace` takes a regex. But, shouldn't the `$` be a `^`?

**Review by toddburnside — APPROVED** [review]
LGTM
